Makefile

```diff
@@ -116,13 +127,64 @@ verify-codegen: download-deps
 	./hack/verify-codegen.sh
 
 # Build the docker image
-docker-build:
+docker-build: vendor
 	docker build -f Dockerfile.dev . -t ${IMG}
 
 # Push the docker image
 docker-push:
 	docker push ${IMG}
 
+.PHONY: e2e-test-dependencies
+e2e-test-dependencies:
+	./scripts/download-e2e-test-dependencies.sh
+
+.PHONY: e2e-test-env
+e2e-test-env: export PATH:=./bin:${PATH}
+e2e-test-env: e2e-test-dependencies
+	kind create cluster
+
+	# TODO get these from the MetalLB resource yaml
+	docker pull metallb/controller:v0.9.6
+	docker pull metallb/speaker:v0.9.6
+	kind load docker-image metallb/controller:v0.9.6
+	kind load docker-image metallb/speaker:v0.9.6
+	./scripts/install-metallb.sh
+
+	docker pull kennethreitz/httpbin
+	kind load docker-image kennethreitz/httpbin
+
+	# TODO collect these from the operator (run the related functions and collect the referenced images)
+	docker pull gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
+	docker pull docker.io/istio/pilot:${ISTIO_VERSION}
+	docker pull docker.io/istio/proxyv2:${ISTIO_VERSION}
+	kind load docker-image gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
+	kind load docker-image docker.io/istio/pilot:${ISTIO_VERSION}
+	kind load docker-image docker.io/istio/proxyv2:${ISTIO_VERSION}
+
+.PHONY: e2e-test-install-istio-operator
+e2e-test-install-istio-operator: export PATH:=./bin:${PATH}
+e2e-test-install-istio-operator: docker-build
+	# TODO build with TEST_RACE_DETECTOR=1 in docker-build
+	kind load docker-image ${IMG}
+	helm install --wait \
+		--set operator.image.repository=${IMAGE_REPOSITORY} \
+		--set operator.image.tag=${TAG} \
+		--create-namespace \
+		--namespace istio-system \
+		istio-operator-e2e-test \
+		deploy/charts/istio-operator/
+
+	# TODO maybe wait until all pods are up and running?
```

Comment:
The `--wait` flag should do this already:
```
--wait                         if set, will wait until all Pods, PVCs, Services, and minimum number of Pods of a Deployment, StatefulSet, or ReplicaSet are in a ready state before marking the release as successful. It will wait for as long as --timeout
```

Reply:
Yes and no: it only waits for the pods installed by helm. Usually, when the istio-operator is ready, a couple of pods in kube-system are still just starting up. It works out fine now, probably because there is a wait in `TestMain` for the cluster to be reachable. Waiting here for all pods might result in a lower load on the cluster when the actual tests start, so it might remove some flakiness.